src/components/LeafletMap.js

```diff
@@ -365,19 +365,19 @@ class LeafletMap extends Component {
 
       //Check max zoom and use checkSameSiteInACluster function helps identify the similar sites in a cluster
       if (currentZoom === mapProperty.maxZoom || this.checkSameSiteInACluster(markerCluster)) {
-        const markersData = markerCluster.map(item => {
-          return item.options.marker;
-        });
-        removeHighlight();
-        siteDropDownHandler(markersData);
-
         const selectedMarkersCluster = this.createCluster(selectedClusterStyle);
+        const markersData = markerCluster.map(item => item.options.marker);
 
         markerCluster.forEach(selectMaker => {
```

Comment:
Not sure why I didn't pick up on this earlier. for..of loops are generally more performant than forEach 

```js
for (let marker of markerCluster) {
  selectedMarkersCluster.addLayer(marker)
}
```

Reply:
This isn't something you need to do right now. This is a small read-only application. But as you're starting to grow into more ambitious work, this or the classic for are things you should reach for over forEach